### PR TITLE
samples: cellular: udp: remove qemu_x86 support

### PR DESCRIPTION
**Pull request by MirkoCovizzi**
* Remove leftover support for qemu_x86
* Refactor and simplify code structure

### CODE REVIEW

File: samples/cellular/udp/src/main.c [hunk 1]
```diff
@@ -100,22 +82,21 @@ static void lte_handler(const struct lte_lc_evt *const evt)
...
 		k_sem_give(&lte_connected);
 		break;
 	case LTE_LC_EVT_PSM_UPDATE:
 		printk("PSM parameter update: TAU: %d s, Active time: %d s\n",
-			evt->psm_cfg.tau, evt->psm_cfg.active_time);
+		       evt->psm_cfg.tau, evt->psm_cfg.active_time);
 		break;
 	case LTE_LC_EVT_EDRX_UPDATE:
 		printf("eDRX parameter update: eDRX: %.2f s, PTW: %.2f s\n",
 		       evt->edrx_cfg.edrx, evt->edrx_cfg.ptw);
```
Comment (lemrey) [inline samples/cellular/udp/src/main.c:95]: `printf`->`printk`

**Review by lemrey — COMMENTED** [review]
A couple of small things, otherwise looks good.

File: samples/cellular/udp/src/main.c (hunk 1 above)
Comment (MirkoCovizzi) [reply samples/cellular/udp/src/main.c:95]: Good catch

**Review by lemrey — APPROVED** [review]

File: samples/cellular/udp/README.rst [hunk 2]
```diff
@@ -136,8 +134,7 @@ Configuration files
 
 The sample provides predefined configuration files for the following development kits:
 
-* :file:`prj.conf` - For nRF9160 DK and Thingy:91
-* :file:`prj_qemu_x86.conf` - For x86 Emulation (QEMU)
+* :file:`prj.conf` - For nRF91 Series and Thingy:91
 
 They are located in :file:`samples/cellular/udp` folder.
```
Comment (eivindj-nordic) [inline samples/cellular/udp/README.rst:139]: nit: I don't think the configuration files section is needed any more, the prj.conf is standard for all samples.

**Review by eivindj-nordic — APPROVED** [review]
Nice cleanup!

**Review by simensrostad — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**